### PR TITLE
Enable feature-gates for otel subcommands

### PR DESCRIPTION
**Pull request by michalpristas**
Added support for [feature gates](https://github.com/open-telemetry/opentelemetry-collector/blob/main/featuregate/README.md) for `otel` and `validate` commands

### How to test

- Build/Package agent
- run `./elastic-agent otel --feature-gates "-telemetry.useOtelForInternalMetrics"`
- as this is `stable` feature gate, you should see error returned saying `useOtelForInternalMetrics` is stable and cannot be disabled
- running `./elastic-agent otel --feature-gates telemetry.useOtelForInternalMetrics` (difference is `-` in front of a gate name) should result in a warning message in output saying it's stable and collector running normally 
- running `./elastic-agent otel -h` needs to include flag in help message

Fixes: #3873

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -65,6 +65,7 @@ require (
 	go.opentelemetry.io/collector/exporter v0.93.0
 	go.opentelemetry.io/collector/exporter/debugexporter v0.93.0
 	go.opentelemetry.io/collector/exporter/otlpexporter v0.93.0
+	go.opentelemetry.io/collector/featuregate v1.0.1
```
Comment (ycombinator) [inline go.mod:68]: Should this version be kept in sync with the other OTel modules?

**Review by ycombinator — DISMISSED** [review]
Minor comment / question about the `featuregate` module's version; otherwise LGTM.

**Comment by ycombinator** [comment]
Actually, one more suggestion: could you add an integration test that proves that the `--feature-gates` flag works?  Something like `./elastic-agent otel --feature-gates foo.bar` returns an error starting with `invalid argument "foo.bar" for "--feature-gates" flag: no such feature gate "foo.bar". valid gates:`?

File: go.mod (hunk 1 above)
Comment (michalpristas) [reply go.mod:68]: good point

**Review by ycombinator — COMMENTED** [review]
Actually, one more suggestion: could you add an integration test that proves that the `--feature-gates` flag works?  Something like `./elastic-agent otel --feature-gates foo.bar` returns an error starting with `invalid argument "foo.bar" for "--feature-gates" flag: no such feature gate "foo.bar". valid gates:`?

File: internal/pkg/agent/cmd/common.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ func NewCommandWithArgs(args []string, streams *cli.IOStreams) *cobra.Command {
 
 	// Init version information contained in package version file
 	err := version.InitVersionError()
-	if err != nil {
+	if isOtel := len(args) > 1 && args[1] == "otel"; !isOtel && err != nil {
```
Comment (ycombinator) [inline internal/pkg/agent/cmd/common.go:46]: Should we even try to initialize the version (previous line) if we are in `otel` mode?

**Review by ycombinator — APPROVED** [review]

**Pull request merged**